### PR TITLE
allow null value for add, set, replace

### PR DESCRIPTION
**Pull request by xinlian12**
Issue: https://github.com/Azure/azure-cosmos-dotnet-v3/issues/2931

Fix:
based on the contract, null is allowed for Add, Set, Replace

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/PatchAsyncTest.java [hunk 1]
```diff
@@ -251,4 +255,35 @@ public void patchInBulk() {
...
+        patchOperations.replace("/" + stringColumnName, null);
+        patchOperations.set("/" + stringColumnName2, null);
+
+        ObjectNode patchedItem = this.container.patchItem(id, new PartitionKey(pkValue), patchOperations, ObjectNode.class).block().getItem();
+        assertThat(patchedItem).isNotNull();
+        assert(patchedItem.get(newStringColumnName)).isNull();
+        assert(patchedItem.get(stringColumnName)).isNull();
+        assert(patchedItem.get(stringColumnName2)).isNull();
+    }
+}
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/PatchAsyncTest.java:289]: nit - add new line.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosPatchOperations.java [hunk 2]
```diff
@@ -276,4 +273,4 @@ public CosmosPatchOperations increment(String path, double value) {
             }
         );
     }
-}
+}
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosPatchOperations.java:276]: nit - add new line

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/PatchTest.java [hunk 3]
```diff
@@ -363,4 +397,4 @@ public static ToDoActivity createRandomToDoActivity(String pk) {
             );
         }
     }
-}
+}
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/PatchTest.java:400]: same

**Review by kushagraThapar — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/PatchTest.java (hunk 3 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/PatchTest.java]: Updated. Thanks

Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/PatchTest.java]: Updated

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosPatchOperations.java (hunk 2 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosPatchOperations.java]: updated

**Review by FabianMeiswinkel — APPROVED** [review]
Thanks!

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/PatchAsyncTest.java (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/PatchAsyncTest.java]: updated

**Comment by xinlian12** [comment]
The failed tests does not relates to the change included in this PR:
image

**Pull request merged**